drafts/Stormless.md

```diff
@@ -0,0 +1,198 @@
+
+```
+service: signup-stormless
+
+provider:
+  name: aws
+  runtime: python2.7
+
+functions:
+  InviteSlack:
+    events:
+      - http:
+          method: POST
+          path: invite
+    stackstorm:
+      action: slack.users.admin.invite
+      input:
+        email:       "{{ input.body.email }}"
+        first_name:  "{{ input.body.first_name }}"
+      config:
+        admin: ${file(env.yml):slack}
+      output:
+        statusCode: 200
+        body: "{{ output }}"
+
+plugins:
+  - serverless-plugin-stackstorm
+```
+
+This is a good time to learn a bit of Serverless. The quick pointers below won't
+reading the docs to understand
+
+I threw in the `events` section here so that we can test the function with REST call to an AWS API Gateway endpoint.
+Note that this default configuration instructs API Gateway to pass the REST POST call with POST body under the `body` key ([details here](TODO-insert-link)). When we POST `{"first_name": "Santa", 
+"email": "santa@mad.russian.xmas.com"}`, the event passed to the Lambda is:
+
+```
+...
+"body": {
+    "first_name": "Santa", 
+    "email": "santa@mad.russian.xmas.com"
+}
+```
+Knowing the input data structure is important to map it to the action input parameters. It's intuitive: `input` represents `event` parameter of [AWS Lambda programming model](http://docs.aws.amazon.com/lambda/latest/dg/python-programming-model-handler-types.html) (BTW should we call it `event`? Vote with a PR!). [Jinja](http://jinja.pocoo.org/) is used to map the inputs; our JavaScript friends who're less familiar
+with this common Python tool find it intuitive enough in simple cases; and Stack-overflow is full of magic Jinga tricks.
+
+Here we obviously map the two parameters from input body to desired action input parameters. 
+
+To keep the config separately, I created a file `env.yml` and put my config parameters there:
+
+```
+# ./env.yml
+# WARNING: Don't commit to Github :)
+slack:
+  admin_token: "xoxs-111111111111-111111111111-111111111111-1111111111"
+  organization: "my-team"
+
+```
+Than I used it in `serverless.yml` like this: `admin: ${file(env.yml):slack}`. It's self-explanatory. 
+
+Optionally, we can also form a function output from action results. Let's keep it simple for now, I'll share tricks later.
+
+Ok, that's it! The function is ready to fly to AWS. But I take it sloooow. First, I'll package it locally. 
+
+```
+sls -package
+```
+The first time takes a long while as this is the time when the plugin sets up itself. It will pull the Docker images from the Hub (oh, did I say you got to be connected? Or we assume internet connection a basic commodity like briefing air and electric power?). It will install StackStorm runners - the code that knows to run StackStorm exchange packs. It will pull the `slack` pack from Exchange. It will install `slack` pack python dependencies. It takes a long time the first time. Good news: it's only the first time. 
```

Comment:
"since plugin need some time to install runtime dependencies"
"did I say you got to be connected" - insert joke about FFC repeal of NN. Let's check where our corporate stands.
"the code that knows to run" is there a word missing?